lib/active_support/cache/memcached_store.rb

```diff
@@ -130,11 +129,11 @@ def cas(name, options = nil)
 
         handle_exceptions(return_value_on_error: false) do
           instrument(:cas, name, options) do
-            @connection.cas(key, expiration(options)) do |raw_value|
-              entry = deserialize_entry(raw_value)
-              value = yield entry.value
+            @connection.cas(key, expiration(options), !options[:raw]) do |raw_value|
+              value = options[:raw] ? raw_value : deserialize_entry(raw_value).value
+              value = yield value
               break true if read_only
-              serialize_entry(Entry.new(value, **options), options)
+              options[:raw] ? value : serialize_entry(Entry.new(value, **options), options)
```

Comment:
This could be cleaned up but I'm just making this PR for discussion for now, so quick & dirty will do.